subtrack.MAUI/Services/SubscriptionService.cs

```diff
@@ -14,5 +15,20 @@ public async Task<IEnumerable<Subscription>> GetSubscriptions()
         {
             return await _context.Subscriptions.ToListAsync();
         }
+
+        public async Task Update(Subscription subscriptionToUpdate)
+        {
+            var sub = await _context.Subscriptions.FindAsync(subscriptionToUpdate.Id);
+
+            if (sub == null) throw new SubscriptionNotFoundException($"Subscription not found.");
```

Comment:
since you already placed the "$" might aswell write "subscription ({id}) was not found"